### PR TITLE
feat: [SRM-11384]: Add metric thresholds for Newrelic

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 124 Issues 
* Code issue 124 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10581)

**Comment by praneshg239** [comment]
image

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts [hunk 1]
```diff
@@ -63,3 +66,37 @@ export interface InitNewRelicCustomFormInterface {
...
 }
+
+export interface NonCustomMetricFields {
+  newRelicApplication:
+    | string
+    | {
+        label: string
+        value: string
+      }
+  metricPacks: TimeSeriesMetricPackDTO[] | undefined
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts:77]: `metricPacks:?: TimeSeriesMetricPackDTO`

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts [hunk 2]
```diff
@@ -63,3 +66,37 @@ export interface InitNewRelicCustomFormInterface {
...
+  formikValues: any
+  setMappedMetrics: React.Dispatch<React.SetStateAction<CustomSelectedAndMappedMetrics>>
+}
+
+export interface GroupedCreatedMetrics {
+  [Key: string]: GroupedMetric[]
+}
+
+export interface MetricThresholdCommonProps {
+  formikValues: any
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts:96]: Adding type for formikValues : Add this to tech depth

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/MetricThresholdProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { MetricThresholdContext } from './MetricThresholds.constants'
+import IgnoreThresholdContent from './components/IgnoreThresholdsContent'
+import type { MetricThresholdCommonProps } from '../../NewRelicHealthSource.types'
+import FailFastThresholdContent from './components/FailFastThresholdsContent'
+
+export default function MetricThresholdProvider(props: MetricThresholdCommonProps): JSX.Element {
+  const { getString } = useStrings()
+  return (
+    <Container margin={{ bottom: 'huge' }} padding="large">
+      <Accordion activeId="advancedAppD">
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/MetricThresholdProvider.tsx:15]: rename advancedAppD to appropriate healthsource

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import { Container } from '@harness/uicore'
+import { TestWrapper } from '@common/utils/testUtils'
+import MetricThresholdProvider from '../MetricThresholdProvider'
+// import MetricThresholdContent from '@cv/pages/health-source/common/MetricThresholds/MetricThresholdsContent'
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx:6]: remove commented code

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+jest.mock('@cv/pages/health-source/common/MetricThresholds/MetricThresholdsContent', () => () => (
+  <Container data-testid="MetricThresholdContent" />
+))
+
+describe('MetricThreshold', () => {
+  test('should render the component', () => {
+    render(
+      <TestWrapper>
+        {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
+        {/* @ts-ignore */}
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx:18]: try not to use ts-ignore

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/components/__tests__/FailFastThresholdsContent.test.tsx [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+          <MetricThresholdContext.Provider value={MockContextValues}>
+            <FailFastThresholdContent />
+          </MetricThresholdContext.Provider>
+        </FormikForm>
+      </Formik>
+    </TestWrapper>
+  )
+}
+
+describe('AppDIgnoreThresholdTabContent', () => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/components/__tests__/FailFastThresholdsContent.test.tsx:25]: rename

**Review by deepesh-ui — DISMISSED** [review]
NewRelic Changes LGTM

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx (hunk 5 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx:18]: Removed

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/components/__tests__/FailFastThresholdsContent.test.tsx (hunk 6 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/components/__tests__/FailFastThresholdsContent.test.tsx:25]: Done

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx (hunk 4 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/__tests__/MetricThresholdProvider.test.tsx:6]: Done

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts:96]: Yes, this we can pick up during tech debts

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.types.ts]: Done

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/MetricThresholdProvider.tsx (hunk 3 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/components/MetricThresholds/MetricThresholdProvider.tsx]: Done

**Review by deepesh-ui — APPROVED** [review]

**Comment by praneshg239** [comment]
Test coverage for Newrelic metric thresholds are above 90% and that can be seen through the screenshot in previous comment that I took from local coverage report. Force merging now as it is failing in the web hooks.

**Pull request merged**